scripts/variantstore/wdl/extract/uber_monitor.py

```diff
@@ -0,0 +1,186 @@
+import argparse
+import sys
+import re
+import os
+
+global MaxCpu
+global MaxMem
+global MaxMemPct
+global MaxDisk
+global MaxDiskPct
+
+def parse_monitoring_log_files(mlog_files, output_file):
+    with open(output_file, 'w') as output:
+        header = f"Total Mem\tMax Mem Used\tMax Mem Used (%)\tTotal Disk\tMax Disk Used\tMax Disk Used (%)\tTask\tShard\tFile\n"
+        output.write(header)
+
+        for mlog_file in mlog_files:
+            if (not os.path.exists(mlog_file)):
+                eprint(f"ERROR: File {mlog_file} does not exist")
+            parse_monitoring_log_file(mlog_file, output)
+
+def parse_monitoring_log_file(mlog_file, output):
+    eprint(f"Parsing: {mlog_file}")
+
+    if (os.stat(mlog_file).st_size == 0):
+        eprint(f"Skipping zero length file")
+        return
+
+    with open(mlog_file) as ml:
+        advance_to_section(ml, "--- General Information")
+
+        line = ml.readline().rstrip()       # #CPU: 16
+        tokens = line.split(": ")
+        if (tokens[0] != '#CPU'):
+            eprint(f"ERROR: Line '{line}' does not look as expected. Is this a monitoring_log file?")
+            sys.exit(1)
+        NumCPUS = tokens[1]
+        eprint(f"Num CPUs: {NumCPUS}")
+
+        line = ml.readline().rstrip()       # Total Memory: 98.25 GiB
+        tokens = line.split(": ")
+        if (tokens[0] != 'Total Memory'):
+            eprint(f"ERROR: Line '{line}' does not look as expected. Is this a monitoring_log file?")
+            sys.exit(1)
+        subtokens = tokens[1].split()
+        if (len(subtokens) != 2):
+            eprint(f"ERROR: Line '{line}' does not look as expected. Is this a monitoring_log file?")
+            sys.exit(1)
+        TotalMemory = float(subtokens[0])
+        TotalMemoryUnits = subtokens[1]
+        eprint(f"Total Memory: {TotalMemory} {TotalMemoryUnits}")
+
+        line = ml.readline().rstrip()       # Total Disk space: 985.000 GiB
+        tokens = line.split(": ")
+        if (tokens[0] != 'Total Disk space'):
+            eprint(f"ERROR: Line '{line}' does not look as expected. Is this a monitoring_log file?")
+            sys.exit(1)
+        subtokens = tokens[1].split()
+        if (len(subtokens) != 2):
+            eprint(f"ERROR: Line '{line}' does not look as expected. Is this a monitoring_log file?")
+            sys.exit(1)
+        TotalDisk = float(subtokens[0])
+        TotalDiskUnits = subtokens[1]
+        eprint(f"Total Disk space: {TotalDisk} {TotalDiskUnits}")
+
+        advance_to_section(ml, "--- Runtime Information")
+
+        global MaxCpu
+        MaxCpu = -100.0
+        global MaxMem
+        MaxMem = -100.0
+        global MaxMemPct
+        MaxMemPct = -100.0
+        global MaxDisk
+        MaxDisk = -100.0
+        global MaxDiskPct
+        MaxDiskPct = -100.0
```

Comment:
couldn't the initializations happen where the variables are defined and this whole block deleted?

Reply:
They need to be initialized here since they are per used per monitoring log fle